stdlib/database/sql/src/main/java/org/ballerinalang/database/sql/statement/AbstractSQLStatement.java

```diff
@@ -427,28 +430,34 @@ protected void cleanupResources(Statement stmt, Connection conn, boolean connect
                 conn.close();
             }
         } catch (SQLException e) {
-            throw new BallerinaException("error in cleaning sql resources: " + e.getMessage(), e);
+            String errorMessage = "error in cleaning sql resources: ";
+            context.setReturnValues(SQLDatasourceUtils.getSQLDatabaseError(context, e, errorMessage));
+            checkAndObserveSQLError(context, errorMessage + e.getMessage());
```

Comment:
I don't think we should be doing this at this level. IMO we should rather propagate this to upper layer and handle at the concrete Statement (eg: SelectStatement, UpdateStatement) level. Otherwise it will be hard to manage. This applies to other relevant places as well.

May be for database modules, we should define two java exception types for the two Ballerina error types and throw them appropriately, then catch at Statement level and create Ballerina errors. WDYT?